ITISFoundation/osparc-simcore
# ✨ Products: Is3705/product groups  (🗃️ DB change)

## What do these changes do?

- Products have an associated group (see ``products.group_id``) 
- Every user belong to one/more product's group with access ``read=False, write=False, delete=False``, i.e. they belong to the product but cannot see any other members of the products  (i.e. belongs but cannot share)
- User is automatically assigned to a product upon registration.

NOTE: @odeimaiz this PR still does NOT change the API, i.e. the ``list_products`` returns product groups as "non-readable" organizations. I would like to introduce the change in the API in a separate PR.

###  🗃️ DB change
 - adds ``products.group_id`` column
 - Defaults to NULL
 - Lazy creation of ``product.group_id`` upon registration of the first user (after these changes)
 - Current users need to be assigned manually! check with @pcrespov @odeimaiz when released.

## Related issue/s

- Implementing #3705


## How to test

- create several products
- add a service to every product
- register a new user with the default product (i.e. priority=0)
- the user has by default 3 groups: everybody groups (1), a product group and primary group 
- change default product
- user CAN login but does not see any


## Checklist


- [x] ``make version-*``
- [x] ``make openapi.json``
- [x] ``cd packages/postgres-database``, ``make setup-commit``, ``sc-pg review -m "my changes"``
